From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat, 8 Aug 2020 12:31:12 +0900
Subject: network: only process non-error message

(cherry picked from commit fb282d4e256a79b19d3299999024e8fafd2ab0a0)
(cherry picked from commit 120064b4a1a34952cc058e30e67bd3ec848f93c0)
---
 src/network/networkd-address.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/networkd-address.c b/src/network/networkd-address.c
index b09d75e..0349ca6 100644
--- a/src/network/networkd-address.c
+++ b/src/network/networkd-address.c
@@ -487,7 +487,7 @@ static int address_remove_handler(sd_netlink *rtnl, sd_netlink_message *m, Link
         r = sd_netlink_message_get_errno(m);
         if (r < 0 && r != -EADDRNOTAVAIL)
                 log_link_message_warning_errno(link, m, r, "Could not drop address");
-        else
+        else if (r >= 0)
                 (void) manager_rtnl_process_address(rtnl, m, link->manager);
 
         return 1;
